homeassistant/components/wallbox/coordinator.py

```diff
@@ -62,6 +63,29 @@
     210: ChargerStatus.LOCKED_CAR_CONNECTED,
 }
 
+_WallboxCoordinatorT = TypeVar("_WallboxCoordinatorT", bound="WallboxCoordinator")
+_P = ParamSpec("_P")
+
+
+def _require_authentication(
+    func: Callable[Concatenate[_WallboxCoordinatorT, _P], Any]
+) -> Callable[Concatenate[_WallboxCoordinatorT, _P], Any]:
+    """Authenticate with decorator using Wallbox API."""
+
+    def require_authentication(
+        self: _WallboxCoordinatorT, *args: _P.args, **kwargs: _P.kwargs
+    ) -> Any:
+        """Authenticate using Wallbox API."""
+        try:
+            self.authenticate()
+            return func(self, *args, **kwargs)
+        except requests.exceptions.HTTPError as wallbox_connection_error:
+            if wallbox_connection_error.response.status_code == HTTPStatus.FORBIDDEN:
+                raise ConfigEntryAuthFailed from wallbox_connection_error
```

Comment:
Shouldn't we start a reauth flow whenever we fail authentication? If this is raised as part of a service call currently no reauth flow is started. It's only `_async_update_data` that will automatically handle `ConfigEntryAuthFailed` and start a reauth flow.